CMakeLists.txt

```diff
@@ -0,0 +1,845 @@
+# !!! The official build system is SConstruct !!!
+# This is an experimental cmake build file for rippled
+#
+# cmake support in rippled. Currently supports:
+#
+#  * unity/nounity debug/release
+#  * running protobuf
+#  * sanitizer builds
+#  * optional release build with assert turned on
+#  * `target` variable to easily set compiler/debug/unity
+#     (i.e. -Dtarget=gcc.debug.nounity)
+#  * gcc/clang/visual studio/xcode
+#  * linux/mac/win
+#  * gcc 4 ABI, when needed
+#  * ninja builds
+#  * check openssl version on linux
+#  * static builds (swd TBD: needs to be tested by building & deploying on different systems)
+#
+# TBD:
+#  * jemalloc support
+#  * count
+#
+# Notes:
+#  * Use the -G"Visual Studio 14 2015 Win64" generator on Windows. Without this
+#    a 32-bit project will be created. There is no way to set the generator or
+#    force a 64-bit build in CMakeLists.txt (setting CMAKE_GENERATOR_PLATFORM won't work).
+#    The best solution may be to wrap cmake with a script.
+#
+#  * It is not possible to generate a visual studio project on linux or
+#    mac. The visual studio generator is only available on windows.
+#
+#  * The visual studio project can be _either_ unity or
+#    non-unity (selected at generation time).  It does not appear possible
+#    to disable compilation based on configuration.
+#
+#  * Language is _much_ worse than python, poor documentation and "quirky"
+#    language support (for example, generator expressions can only be used
+#    in limited contexts and seem to work differently based on
+#    context (set_property can set multiple values, add_compile_options
+#    can not/or is buggy)
+#
+#  * Could not call out to `sed` because cmake messed with the regular
```

Comment:
ok, so here is an example of some CMake code I wrote to extract and parse a tag...it might be a starting point

    find_package(Git)
    set(${VERSION_BUILD_MINOR} "0")
    if(GIT_FOUND)
        # Use 'git describe --long' to find the distance from the beginning of the branch
        execute_process(COMMAND ${GIT_EXECUTABLE} "describe" "--long"
                        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
                        RESULT_VARIABLE GIT_EXIT_CODE
                        OUTPUT_VARIABLE TAG_BEGIN_BRANCH
                        ERROR_QUIET)
        if(GIT_EXIT_CODE EQUAL 0)
            # Example git output: tag_begin_eagle-10-gcd06d86
            # Extact the distance digit (10 in this case)
            if(${TAG_BEGIN_BRANCH} MATCHES "^tag_begin_.+-([0-9]+)-.*$")
                set(${VERSION_BUILD_MINOR} ${CMAKE_MATCH_1})
            endif()
        endif()
    endif(GIT_FOUND)